sim-cli/Cargo.toml

```diff
@@ -25,6 +25,7 @@ rand = "0.8.5"
 hex = {version = "0.4.3"}
 futures = "0.3.30"
 console-subscriber = { version = "0.4.0", optional = true}
+triggered = "0.1.2"
```

Comment:
I think it is not necessary to add this dependency to this PR. Maybe just focus on the refactor part first.